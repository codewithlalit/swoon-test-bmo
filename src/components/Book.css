.card {
  width: 200px;
  height: auto;
  min-height: 300px;
  padding: 5px;
  margin: 10px;
  background-color: rgb(241, 227, 227);
}
img {
  border: 1px solid #ddd;
  border-radius: 4px;
  /* padding: 5px; */
  width: 150px;
  height: 150px;
}
