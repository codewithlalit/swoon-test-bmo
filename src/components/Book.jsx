import React from "react";
import "./Book.css";

const Book = ({ book }) => {
  let coverPic;

  let cover = book.isbn ? book.isbn[0] : null;
  if (cover === null) {
    coverPic = null;
  } else {
    coverPic = `https://covers.openlibrary.org/b/isbn/${cover}-M.jpg`;
  }
  // console.log(cover);

  return (
    <div className='card'>
      <h5>{book.title}</h5>
      <h6>Author: {book.author_name}</h6>
      <img src={coverPic} alt='book' />
      <p>
        Publish date: {book.first_publish_year ? book.first_publish_year : null}
      </p>
    </div>
  );
};

export default Book;
