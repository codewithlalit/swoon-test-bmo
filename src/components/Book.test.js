import renderer from "react-test-renderer";
import Book from "./Book";

const book = { isbn: [] };

it('Book does have isbn', ()=>{
    expect(book).toHaveProperty('isbn')
})

it('renders a snapshot',()=>{
    const tree = renderer.create(<Book book={book}/>).toJSON()
    expect(tree).toMatchSnapshot();
})