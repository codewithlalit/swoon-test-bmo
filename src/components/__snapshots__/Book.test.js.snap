// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div
  className="card"
>
  <h5 />
  <h6>
    Author: 
  </h6>
  <img
    alt="book"
    src="https://covers.openlibrary.org/b/isbn/undefined-M.jpg"
  />
  <p>
    Publish date: 
  </p>
</div>
`;
